src/libraries/System.Console/tests/ConsoleEncoding.Windows.cs

```diff
@@ -38,6 +38,27 @@ public void InputEncoding_SetUnicodeEncoding_SilentlyIgnoredInternally()
         }).Dispose();
     }
 
+    [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public void InputEncoding_SetEncodingWhenDetached_ErrorIsSilentlyIgnored()
+    {
+        RemoteExecutor.Invoke(() =>
+        {
+            Encoding encoding = Encoding.GetEncoding(0);
+            Assert.NotEqual(encoding.CodePage, Console.InputEncoding.CodePage);
+
+            // use FreeConsole to detach the current console - simulating a process started with the "DETACHED_PROCESS" flag
+            FreeConsole();
+
+            // Setting the input encoding should not throw an exception
+            Console.InputEncoding = encoding;
+            // The internal state of Console should have updated, despite the failure to change the console's input encoding
+            Assert.Equal(encoding, Console.InputEncoding);
+            // Operations on the console are no longer valid - GetConsoleCP fails.
+            Assert.Equal(0u, GetConsoleCP());
```

Comment:
I would advise testing that Console.WriteLine() still works here. The caller would need to use StandardOutputRedirected = true for the test case.